docs/customization.md

```diff
@@ -1,38 +1,53 @@
 ## Customization
-### Modifying the client library format
 
-Don't like the default client syntax?  Want a different language supported?  No problem!  OpenAPI Generator processes mustache templates with the [jmustache](https://github.com/samskivert/jmustache) engine.  You can modify our templates or make your own.
+### Modifying a template
 
-You can look at `modules/openapi-generator/src/main/resources/${your-language}` for examples. To make your own templates, create your own files and use the `-t` flag to specify your template folder.  It actually is that easy.
+Clone OpenAPI Generator and take a look at the following directory: `modules/openapi-generator/src/main/resources/${template}`. In here you'll see all of the generators available, for most programming languages, web application frameworks and web servers.
 
-### Making your own codegen modules
+Templates consist of multiple mustache files. Mustache is a simple, readable templating language, and the engine used is [jmustache](https://github.com/samskivert/jmustache).
 
-If you're starting a project with a new language and don't see what you need, openapi-generator can help you create a project to generate your own libraries:
+If you wish to modify one of these templates, copy and paste the template you're interested in to a templates directory you control. To let OpenAPI Generator know where this templates directory is, use the `-t` option (e.g: `-t ./templates/`).
+
+To tie that all together:
+
+```sh
+mkdir templates
+cp -r modules/openapi-generator/src/main/resources/${template} templates/
+java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar generate \
+  -t ./templates/ -g ruby -i ./foo.yml -o ./out/ruby
+```
+
+_**Note:** You cannot use this approach to create new templates, only override existing ones._
+
+### Creating a new template
+
+If none of the templates suit your needs at all, you can create a brand new template. OpenAPI Generator can help with this, using the `meta` command:
 
 ```sh
 java -jar modules/openapi-generator-cli/target/openapi-generator-cli.jar meta \
-  -o output/myLibrary -n myClientCodegen -p com.my.company.codegen
+  -o codegens/customCodegen -n myTemplate -p com.my.company.codegen
```

Comment:
I don't know that `codegens/customCodegen` is any better than `output/myLibrary`, and I worry that it might lead to confusion.  `codegens` as a directory may make it seem like this is some conventional or required directory name, while `output` is commonly understood as just a placeholder for your output. This could probably be resolved if you add that the output directory can be anything?

Also, `-n myTemplate` is not the convention we normally use, and yet `myClientCodegen` is rather outdated (I think since swagger-codegen 2.0.0 or so). `MyTemplate` as a class name doesn't make sense here because you're generating the codegen type and not a template type.

Compare your output:

```
$ tree codegens/
codegens/
└── customCodegen
    ├── README.md
    ├── pom.xml
    └── src
        └── main
            ├── java
            │   └── com
            │       └── my
            │           └── company
            │               └── codegen
            │                   └── MytemplateGenerator.java
            └── resources
                ├── META-INF
                │   └── services
                │       └── org.openapitools.codegen.CodegenConfig
                └── myTemplate
                    ├── api.mustache
                    ├── model.mustache
                    └── myFile.mustache

12 directories, 7 files
```

With the previous:

```
$ tree output/
output/
└── myLibrary
    ├── README.md
    ├── pom.xml
    └── src
        └── main
            ├── java
            │   └── com
            │       └── my
            │           └── company
            │               └── codegen
            │                   └── MyclientcodegenGenerator.java
            └── resources
                ├── META-INF
                │   └── services
                │       └── org.openapitools.codegen.CodegenConfig
                └── myClientCodegen
                    ├── api.mustache
                    ├── model.mustache
                    └── myFile.mustache

12 directories, 7 files
```

If `-n` is provided something equivalent to what you would pass as `--generator-name` (previously `--lang`), you would get a reasonable output:

```
$ tree output/
output/
└── finatra-server
    ├── README.md
    ├── pom.xml
    └── src
        └── main
            ├── java
            │   └── us
            │       └── jimschubert
            │           └── generators
            │               └── FinatraGenerator.java
            └── resources
                ├── META-INF
                │   └── services
                │       └── org.openapitools.codegen.CodegenConfig
                └── finatra
                    ├── api.mustache
                    ├── model.mustache
                    └── myFile.mustache

11 directories, 7 files
```

Compare this to the files generated by the `new.sh` script I have under the project root, which is intended for bootstrapping new generators within our project:

```
./new.sh -n finatra -s -t
```

You'll notice that the output from `new.sh` and the output from the last command above are relatively similar. That is, it would be much easier for someone to contribute their custom generated code back to the project following those conventions (namely `-n` matching generator-name/lang formats).